src/components/KTreeList/KTreeList.vue

```diff
@@ -0,0 +1,170 @@
+<template>
+  <div class="k-tree-list">
+    <KTreeDraggable
+      :items="internalList"
+      :disable-drag="disableDrag"
+      :max-level="maxLevel"
+      @change="handleChangeEvent"
+      @child-change="handleChildChangeEvent"
+      @selected="handleSelection"
+    >
+      <template #item-icon="{ item }">
+        <slot
+          :item="item"
+          name="item-icon"
+        />
+      </template>
+      <template #item-label="{ item }">
+        <slot
+          :item="item"
+          name="item-label"
+        />
+      </template>
+    </KTreeDraggable>
+  </div>
+</template>
+
+<script lang="ts" setup>
+/**
+ *  Note: if TS errors appear on the
+ *  <template #[itemIcon]="slotProps"> or
+ *  <template #[itemLabel]="slotProps">
+ * lines switch to `v-slot:` notation and Save to let linter clear -->
+ */
+import { computed, ref, watch, onMounted, PropType } from 'vue'
+import KTreeDraggable from '@/components/KTreeList/KTreeDraggable.vue'
+import { TreeListItem } from '@/components/KTreeList/KTreeItem.vue'
+
+export interface ChangeEvent {
+  items: TreeListItem[]
+}
+
+export interface ChildChangeEvent {
+  parent: string,
+  children: TreeListItem[]
+}
+
+const props = defineProps({
+  modelValue: {
+    type: Array as PropType<TreeListItem[]>,
+    default: null,
+    validator: (items: TreeListItem[]) => !items.length || items.every(i => i.name !== undefined && i.id !== undefined),
+  },
+  items: {
+    type: Array as PropType<TreeListItem[]>,
+    default: null,
+    validator: (items: TreeListItem[]) => !items.length || items.every(i => i.name !== undefined && i.id !== undefined),
+  },
+  disableDrag: {
+    type: Boolean,
+    default: false,
+  },
+  maxLevel: {
```

Comment:
Kind of a nit: I think this prop name should be plural
```suggestion
  maxLevels: {
```